File: LeetCode/Medium/0347-top-k-frequent-elements/0347-top-k-frequent-elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function (nums, k) {
  let obj = {};
  for (const num of nums) {
    // #1. 없음
    if (!obj[num]) {
      obj[num] = 1;
    }
    // #2. 있음
    else {
      obj[num] += 1;
    }
  }
  let arr = Object.entries(obj).sort((a, b) => b[1] - a[1]);
  console.log(arr);
  const res = [];
  for (let i = 0; i < k; i++) {
    res.push(parseInt(arr[i][0]));
  }
  return res;
};

File: LeetCode/Medium/0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        #Example
        # Input: nums = [1, 1, 1, 2, 2, 3], k = 2
        # Output: [1, 2]
        set = {}
        for num in nums:
            set[num] = set.get(num,0) + 1
        carr = []
        for key,value in set.items():
            carr.append([key,value])
        carr.sort(reverse=True, key=lambda x: x[1])
        
        answer = []
        cnt = 0
        while(cnt < k):
            answer.append(carr[cnt][0])
            cnt += 1
        return answer